core/src/main/java/org/apache/gravitino/catalog/FilesetNormalizeDispatcher.java

```diff
@@ -96,7 +96,9 @@ public boolean dropFileset(NameIdentifier ident) {
 
   @Override
   public String getFileLocation(NameIdentifier ident, String subPath) {
-    throw new UnsupportedOperationException("Not implemented");
+    // The constraints of the name spec may be more strict than underlying catalog,
+    // and for compatibility reasons, we only apply case-sensitive capabilities here.
+    return dispatcher.getFileLocation(normalizeCaseSensitive(ident), subPath);
```

Comment:
I would suggest split the core part logic into another PR. In this PR, we only focus on client and server side API. Don't make the PR too big, which is hard to review.

Reply:
Done, this PR has been split to only include changes with the Java Client and Server.